src/v/redpanda/admin/transform.cc

```diff
@@ -269,3 +274,74 @@ admin_server::garbage_collect_committed_offsets(
     co_await throw_on_error(*req, ec, model::controller_ntp);
     co_return ss::json::json_void();
 }
+
+model::transform_metadata_patch
+parse_json_metadata_patch(const json::Document& doc) {
+    model::transform_metadata_patch result;
+
+    if (doc.HasMember("env")) {
+        if (!doc["env"].IsArray()) {
+            throw ss::httpd::bad_request_exception(
+              "Field 'env' must be an array");
+        }
+        result.env.emplace();
+        absl::c_transform(
+          doc["env"].GetArray(),
+          std::inserter(result.env.value(), result.env.value().end()),
+          [](const auto& p) {
+              if (!p.IsObject()) {
+                  throw seastar::httpd::bad_request_exception(
+                    "Env array items are not objects");
+              }
+              for (std::string_view k : {"key", "value"}) {
+                  if (!p.HasMember(k.data()) || !p[k.data()].IsString()) {
+                      throw seastar::httpd::bad_request_exception(
+                        fmt::format("Env array item missing field '{}'", k));
+                  }
+              }
+              return std::make_pair(
+                p["key"].GetString(), p["value"].GetString());
+          });
+    }
+
+    if (doc.HasMember("is_paused")) {
+        if (!doc["is_paused"].IsBool()) {
+            throw ss::httpd::bad_request_exception(
+              "Field 'is_paused' must be a boolean");
+        }
+        result.paused.emplace(doc["is_paused"].GetBool());
+    }
+
+    return result;
+}
+
+ss::future<ss::json::json_return_type>
+admin_server::patch_transform_metadata(std::unique_ptr<ss::http::request> req) {
+    if (!_transform_service->local_is_initialized()) {
+        throw transforms_not_enabled();
+    }
+
+    ss::sstring raw_name;
+    if (!admin::path_decode(req->param["name"], raw_name)) {
```

Comment:
Fyi, after the [seastar upgrade](https://github.com/redpanda-data/redpanda/pull/17899/commits/45ba8c941c511628907ea7d13234c8b75c2838d8) we should use `req->get_path_param()` here instead. I think this would show a deprecation warning if you rebase to `dev` now.

Reply:
It does indeed. very nice 👌